eva_cttv_pipeline/evidence_string_generation/pipeline.nf

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl=2
+
+
+def helpMessage() {
+    log.info"""
+    Generate ClinVar evidence strings for Open Targets.
+    
+    Params:
+        --batch_root    Directory for current batch
+        --schema        Open Targets JSON schema version
+        --clinvar       ClinVar XML file (optional, will download latest if omitted)
+    """
+}
+
+params.help = null
+params.batch_root = null
+params.schema = null
+params.clinvar = null
+
+if (params.help) {
+    exit 0, helpMessage()
+}
+if (!params.batch_root || !params.schema) {
+    exit 1, helpMessage()
+}
+batchRoot = params.batch_root
+
+
+/*
+ * Main workflow.
+ */
+workflow {
+    if (params.clinvar != null) {
+        clinvarXml = Channel.fromPath(params.clinvar)
+    } else {
+        clinvarXml = downloadClinvar()
+    }
+    downloadJsonSchema()
+
+    runSnpIndel(clinvarXml)
+    runRepeat(clinvarXml)
+    runStructural(clinvarXml)
+    combineConsequences(runSnpIndel.out.consequencesSnp,
+                        runRepeat.out.consequencesRepeat,
+                        runStructural.out.consequencesStructural)
+    
+    generateEvidence(clinvarXml,
+                     downloadJsonSchema.out.jsonSchema,
+                     combineConsequences.out.consequencesCombined)
+    checkDuplicates(generateEvidence.out.evidenceStrings)
+
+    convertXrefs(clinvarXml)
+}
+
+/*
+ * Download ClinVar data, using the most recent XML dump.
+ */
+process downloadClinvar {
+    output:
+    path "clinvar.xml.gz", emit: clinvarXml
+
+    script:
+    """
+    wget -O clinvar.xml.gz \
+        https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarFullRelease_00-latest.xml.gz
+    """
+}
+
+/*
+ * Download the Open Targets JSON schema.
+ */
+process downloadJsonSchema {
+    output:
+    path "opentargets-${params.schema}.json", emit: jsonSchema
+
+    script:
+    """
+    wget -O opentargets-${params.schema}.json \
+        https://raw.githubusercontent.com/opentargets/json_schema/${params.schema}/opentargets.json
+    """
+}
+
+/*
+ * Run simple variants (SNPs and other variants with complete coordinates) through VEP and map them
+ * to genes and functional consequences.
+ */
+process runSnpIndel {
```

Comment:
The original run_consequence_mapping.sh was set with -M 10G for memory requirement.
Is this not required ?

Reply:
Interesting, apparently not... just double checked the logs on the cluster and it indeed ran successfully with the default 4G memory.